.github/workflows/rust.yml

```diff
@@ -419,3 +419,38 @@ jobs:
 
       - if: "failure()"
         uses: "andymckay/cancel-action@271cfbfa11ca9222f7be99a47e8f929574549e0a" # v0.4
+
+  no-std-tests:
+    name: "Test (no_std)"
+    runs-on: ubuntu-latest
+    needs: [machete, docs]
+    timeout-minutes: 30
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      # Note: needs nighly toolchain because otherwise we cannot define custom lang-items.
+      - name: Install Rust nightly toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          override: true
+
+      - name: Add the `thumbv7em-none-eabi` target
+        run: |
+          rustup target add `thumbv7em-none-eabi`
+
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@23bce251a8cd2ffc3c1075eaa2367cf899916d84 # v2.7.3
+
+      # Note: We currently do not have a way to run real tests in a `no_std` environment.
+      # We can only make sure that they compile to ARM thumb ISA.
+      # Running the binary and inspecting the output would require an actual machine with matching ISA or some sort of emulator.
+      - name: Compile `no-std-tests` crate to `thumbv7em-none-eabi` target.
+        run: |
+          cargo build --target thumbv7em-none-eabi
```

Comment:
huh,  you don't need `sudo apt install gcc-arm-none-eabi`?

Reply:
I want to see if it works in the CI without it first, for the `subxt-metadata` crate alone I remember if worked on my machine before installing `gcc-arm-none-eabi`, My guess is that it is only required for some things that the core and signer crate are doing. Let's see...